### DISCUSSION TITLE
Timeout error awaiting page resources

### DISCUSSION THREAD

**Discussion by zilch42** [zauberzeug/nicegui#2429] [Q&A] [answered]
### Question

I have a page which needs to load some resources in order to populate the UI, however when that loading takes too long (even if it is awaited), I get a `TimeoutError: Response not ready after 3.0 seconds`, I'm guessing because it still needs to wait for the async tasks to complete before returning the page? 

```python
from nicegui import ui
import asyncio

@ui.page('/')
def home():
    ui.label("home")
    ui.link("Loading Page", "/load")

@ui.page('/load')
async def loading_page():
    # data = await run.io_bound("load big dataset")
    # do UI stuff with data
    await asyncio.sleep(5)
    ui.label("That took a while ")

ui.run()
```

What would be a better way? Can I load the page, maybe with a spinner showing that stuff is happening, before all of the io_bound stuff is completed? All I can think to do at the moment is put the loading task behind a button, but that's not ideal. 

As a stop gap measure, I tried setting `ui.run(reconnect_timeout=100)`, but that doesn't seem to have any effect on the 3.0 second time out (and wouldn't be the ideal solution anyway).

**Comment by rodja** [comment, marked as answer]
You can await connection and then do the heavy computation async:

```py
@ui.page('/load')
async def loading_page(client: Client):
    spinner = ui.spinner('bars', size='8em')
    await client.connected()
    await asyncio.sleep(5)
    spinner.visible = False
    ui.label("That took a while ")
```

**Reply by zilch42** [reply]
That's done it! Thanks!
